### PR TITLE
feat: upgrade api version to v10

### PR DESCRIPTION
**Pull request by Azn9**
Behavior change:
- To access the content of `AutoModActionExecutedEvent#getContent` and `AutoModActionExecutedEvent#getMatchedContent`, the `MESSAGE_CONTENT` intent is required

### CODE REVIEW

**Comment by Doc94** [comment]
LGTM only mention in the docs about the exception.

File: core/src/main/java/discord4j/core/event/domain/automod/AutoModActionExecutedEvent.java [hunk 1]
```diff
@@ -188,7 +189,20 @@ public Optional<Snowflake> getSystemMessageId() {
      * @return The contents.
      */
     public String getContent() {
```
Comment (Doc94) [inline core/src/main/java/discord4j/core/event/domain/automod/AutoModActionExecutedEvent.java:191]: This require mention the throw in docs

File: core/src/main/java/discord4j/core/event/domain/automod/AutoModActionExecutedEvent.java [hunk 2]
```diff
@@ -206,6 +220,19 @@ public Optional<String> getMatchedKeyword() {
      * @return The matched content, if present.
      */
     public Optional<String> getMatchedContent() {
```
Comment (Doc94) [inline core/src/main/java/discord4j/core/event/domain/automod/AutoModActionExecutedEvent.java:222]: This require mention the throw in docs

File: rest/src/main/java/discord4j/rest/route/Routes.java [hunk 3]
```diff
@@ -29,7 +29,7 @@ public abstract class Routes {
      * @see <a href="https://discord.com/developers/docs/reference#base-url">
      * https://discord.com/developers/docs/reference#base-url</a>
      */
-    public static final String BASE_URL = "https://discord.com/api/v9";
+    public static final String BASE_URL = "https://discord.com/api/v10";
```
Comment (Doc94) [inline rest/src/main/java/discord4j/rest/route/Routes.java:32]: This change not break any more than the intent?

**Review by Doc94 — CHANGES_REQUESTED** [review]

File: rest/src/main/java/discord4j/rest/route/Routes.java (hunk 3 above)
Comment (Azn9) [reply rest/src/main/java/discord4j/rest/route/Routes.java:32]: There are other changes but already supported with the current version of D4J (as the gateway already uses the latest version)

**Pull request merged**